src/base/rss/rssfolder.cpp

```diff
@@ -189,7 +189,6 @@ QHash<QString, FeedPtr> Folder::getAllFeedsAsHash() const
 
 bool Folder::addFile(const FilePtr &item)
 {
-    Q_ASSERT(!m_children.contains(item->id()));
```

Comment:
Because it seemed unnecessary and gets triggered in certain cases with my changes. The logic right below it is already wrapped in the same condition, so having the assert as well just reduces the usefulness of this function.